### ISSUE TITLE
Translation - Portuguese

### ISSUE THREAD

**Issue by Benjamin-Waters** [Benjamin-Waters/ScepterOfLife#69]

**Comment by hi15** [comment]
Translate what? You can explain please?

**Comment by Benjamin-Waters** [comment]
This is an add-on campaign for a video game called [The battle for Wesnoth](http://wesnoth.org/). While this add-on worked for version 1.6 of wesnoth, it no longer works for 1.13 and the original author seems to have abandoned it.

One of the things that was never done for it was translations. So for each of the 12 Scenarios, there is quite a bit of dialogue. For example here is some from Scenario 1:

#define SOL_01_OPENING_DIALOGUE_1
#ifndef SOL_01_NO_CONVERSE
    [message]
        id={SOL_SMYRNA}
        message= _ "Hail, Aeryn! Where is your brother? I desperately need to speak with him."
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "He has been called away to battle the orcs in the east."
    [/message]
    [message]
        id={SOL_SMYRNA}
        message= _ "And what of the town defenses?"
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "They are with him. What urgent business requires our soldiers and leader?"
    [/message]
    [message]
        id={SOL_SMYRNA}
        message= _ "Orcs have taken the northern pass, and are marching toward us as we speak."
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "What?! How much time do we have?"
    [/message]
    [message]
        id={SOL_SMYRNA}
        message= _ "Not long."
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "..."
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "Okay, Smyrna - ride east; find my brother. I and the townsmen will hold the defenses until he returns."
    [/message]
    [message]
        id={SOL_SMYRNA}
        message= _ "Very well, Aeryn. Your brother is likely fighting on the main front, so expect his return in 2 days. Once he arrives, you'll be able to recruit some real soldiers."
    [/message]
    [message]
        id={SOL_AERYN}
        message= _ "Then I wish you godspeed."
    [/message]
#endif
#enddef


It would be nice to have this campaign translated into all the languages Wesnoth is translated in. Wesnoth currently has Portuguese as an official language (both Brazil and Portugal locals). If any one was willing to translate all the dialogue text from the scenarios from English to either of the Portuguese locals, I would make sure that translation was in the add-on.

Does that help clarify?

**Comment by hi15** [comment]
Yes, thanks. You have an example of pattern that I need follow? (If I just replace english for porguese, or write in a new "message= _ " below english version, or create a new file like the existent in scenarios)

**Comment by Benjamin-Waters** [comment]
It seems like [getText](https://en.wikipedia.org/wiki/Gettext) is what's used for the translations so if you're feeling adventurous you could try looking into that, I've seen in other add-ons they ultimately end up with a .mo file that the game uses.

Otherwise just having the text in txt files for the content could be done and then we can look into the technical details later.
